Neutral options

saarasat opened the pull request:
# Description

Fixes: https://github.com/apluslms/mooc-grader/issues/55 

Neutral options of checkbox-type questions have been added to the model answer in questionnaire. Previously they only showed briefly at the moment when the student reached maximum amount of submissions if the “reveal-model-at-max-submissions" was set to true. 

At the same time, the role of these two different model answers have been cleared up a bit and pushed towards the other [general exercise configuration](https://github.com/apluslms/mooc-grader/tree/master/courses#accesstypesstdsynccreateform), by renaming the form attributes for the model answers.

The only visible change to appearance, is the indication of neutral answers and hiding the option-related hints, when checking the model answer:

![image](https://user-images.githubusercontent.com/43036333/89891647-1e8cfb80-dbde-11ea-93ad-4ea1a473f2ef.png)

# Testing 

Has been tested with: 

1. Normal checkboxes without neutral option(s) 
2. Normal checkboxes with neutral option(s) 
3. Randomized checkboxes with neutral option(s) 
4. Randomized checkboxes without neutral option(s) 
5. Questionnaires with other types of questions at the same time as checkbox-questions (radio-, text- and dropdown-fields)
6. All the above, while there are general and option-related hints available 

Existing automatic tests also pass.

# Have you updated the README or other relevant documentation?

Not really relevant

# Is it [Done](https://wiki.aalto.fi/display/EDIT/Definition+of+Done)?

- [ ] I (developer) have created unit tests and the tests pass
- [ ] I (developer) have created functional tests (Selenium tests) if applicable
- [x] I (developer) have tested the changes manually
- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

raphendyr commented on access/types/forms.py:
```diff
@@ -33,18 +33,18 @@ def __init__(self, *args, **kwargs):
-        self.show_correct = kwargs.pop('show_correct') if 'show_correct' in kwargs else False
-        self.show_correct_once = kwargs.pop('show_correct_once') if 'show_correct_once' in kwargs else False
+        self.model_answer = kwargs.pop('model_answer') if 'model_answer' in kwargs else False
+        self.reveal_correct = kwargs.pop('reveal_correct') if 'reveal_correct' in kwargs else False
```
`self.reveal_correct = kwargs.pop('reveal_correct', False)` should be sufficient

saarasat replied:
Very true, I changed those.

raphendyr commented on access/types/stdsync.py:
```diff
@@ -151,11 +151,12 @@ def createForm(request, course, exercise, post_url):
     result = { "form": form, "accepted": True, "points": points,
-        "error_groups": error_groups, "error_fields": error_fields }
+        "error_groups": error_groups, "error_fields": error_fields,
+        "model_answer": True }
```
This would be a good time to move every key on it's own line.

Mankro commented on access/types/forms.py:
```diff
@@ -323,26 +320,29 @@ def create_more(self, configuration):
                 choices.append((value, mark_safe(label)))
+                if opt.get('correct') == "neutral":
+                    neutral.append(value)
                 if opt.get('correct', False) is True:
```
`elif` could be used since `opt.get('correct')` can not be both `"neutral"` and `True` at the same time. Now, two `if` conditions are always checked.

saarasat replied:
You are correct! I changed that now

Mankro commented on locale/fi/LC_MESSAGES/django.po:
```diff
@@ -332,21 +332,25 @@ msgstr[1] ""
-#: access/templates/access/graded_form_feedback.html:15
+#: access/templates/access/graded_form.html:134
+msgid "neutral option which does not affect grading"
+msgstr "vaihtoehto ei vaikuta kokonaispisteisiin"
```
I wonder if the Finnish string should include "neutral": "neutraali vaihtoehto ei vaikuta kokonaispisteisiin".

saarasat replied:
Added that as well, made sense to me.

saarasat commented:
This one is also now squashed, tested, and rebased on top of the current master (https://github.com/apluslms/mooc-grader/commit/0df9b3b197171396ec97dbab2b3481b6a494422a). Also translations have been checked.
